node/cn/api_tracer_test.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2019 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package cn
+
+import (
+	"context"
+	"github.com/golang/mock/gomock"
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/blockchain/vm"
+	"github.com/klaytn/klaytn/common/hexutil"
+	mocks2 "github.com/klaytn/klaytn/consensus/mocks"
+	"github.com/klaytn/klaytn/networks/rpc"
+	mocks3 "github.com/klaytn/klaytn/node/cn/mocks"
+	"github.com/klaytn/klaytn/work/mocks"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func createCNMocks(t *testing.T) (*gomock.Controller, *PrivateDebugAPI, *mocks2.MockEngine, *mocks.MockBlockChain, *mocks3.MockMiner) {
+	mockCtrl, mockEngine, mockBlockChain, _ := newMocks(t)
+	mockMiner := mocks3.NewMockMiner(mockCtrl)
+	api := NewPrivateDebugAPI(nil, &CN{miner: mockMiner, blockchain: mockBlockChain, engine: mockEngine})
+	return mockCtrl, api, mockEngine, mockBlockChain, mockMiner
+}
+
+func TestPrivateDebugAPI_TraceChain(t *testing.T) {
+	endBlockNumber := rpc.BlockNumber(123)
+	block := newBlock(123)
+	{
```

Comment:
These all error cases covers only `from = nil`.
Why don't you add `from != nil && to == nil` case?

Reply:
I thought I did it =( I will modify the test to cover that case. Thanks!